app/vmselect/netstorage/tenant_cache.go

```diff
@@ -17,28 +17,36 @@ import (
 )
 
 var (
-	tenantsCacheDuration = flag.Duration("search.tenantCacheExpireDuration", 5*time.Minute, "The expiry duration for list of tenants for multi-tenant queries.")
+	tenantsCacheDuration = flag.Duration("search.tenantCacheExpireDuration", 5*time.Minute, "The expiry duration for list of tenants for multi-tenant queries. 0 value disables caching of tenants.")
```

Comment:
> Zero value disables caching for tenants list

I think we should reprhase it, because I read it: `A user can provide a list of tenants for which the cache would be disabled`. But that's not the case.



```suggestion
	tenantsCacheDuration = flag.Duration("search.tenantCacheExpireDuration", 5*time.Minute, "Expiry duration for caching tenants in memory. A zero value disables caching, causing tenants to be fetched from storage nodes on every query.")
```